From: Stefano Rivera <stefanor@debian.org>
Date: Sat, 13 Nov 2021 16:36:30 -0800
Subject: C Tests: Avoid creating a 0-length array
MIME-Version: 1.0
Content-Type: text/plain; charset="utf-8"
Content-Transfer-Encoding: 8bit

C does not permit 0-length arrays, triggering this warning (raised to an
error with -Werror) from GCC 11:

```
build/test_poly1305_load_m.c:16:5: error: ‘poly1305_load_m’ reading 1 byte from a region of size 0 [-Werror=stringop-overread]
   16 |     poly1305_load_m(m, secret, 0);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Forwarded: https://github.com/Legrandin/pycryptodome/pull/572
---
 src/test/make_tests_poly1305_load_m.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/test/make_tests_poly1305_load_m.py b/src/test/make_tests_poly1305_load_m.py
index 3691dda..fa6915f 100644
--- a/src/test/make_tests_poly1305_load_m.py
+++ b/src/test/make_tests_poly1305_load_m.py
@@ -35,7 +35,7 @@ print("#include <stdio.h>")
 print()
 print("void poly1305_load_m(uint32_t r[5], const uint8_t data[], size_t len);")
 
-for len_secret in range(16+1):
+for len_secret in range(1, 16+1):
     make_test(b"\xaa" * len_secret)
 make_test(b"\xcc" * 16)
 make_test(b"\xff" * 16)
